Enhance purview commands with check on application permissions

MathijsVerbeeck opened the pull request:
Closes #4450

milanholemans commented:
Well done @MathijsVerbeeck, we'll try to review it ASAP!

milanholemans reviewed (changes requested):
Great work so far, what is included in this PR is looking good for me, but I have a few remarks.

Seems like we are missing the command: `purview retentionevent remove`.

Seems like following commands throw a different error, I suggest that we update them to the error you are using to make it more unfied.

- retentionevent get
- retentionevent list 
- retentioneventtype list

Also in these commands, the check is done deeper inside the command. I suggest that we move to the top so this is the first thing that happens when running the command. The approach you are using is the right one 👍

MathijsVerbeeck commented:
Hi @milanholemans 

I added the check on `purview retentionevent remove`. I also changed the checks on `retentionevent get / list` and `retentioneventtype list`.

milanholemans reviewed (approved):
Looks good to me!

milanholemans commented:
Merged manually. Thank you for this great addition! 👏
